contrib/pkg/createcluster/azure.go

```diff
@@ -0,0 +1,70 @@
+package createcluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/sirupsen/logrus"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1alpha1"
+	hivev1azure "github.com/openshift/hive/pkg/apis/hive/v1alpha1/azure"
+)
+
+const (
+	azureCredFile = "osServicePrincipal.json"
+)
+
+var _ cloudProvider = (*azureCloudProvider)(nil)
+
+type azureCloudProvider struct {
+}
+
+func (p *azureCloudProvider) generateCredentialsSecret(o *Options) (*corev1.Secret, error) {
+	spFile := filepath.Join(os.Getenv("HOME"), ".azure", azureCredFile)
+	log.Infof("Loading Azure service principal from: %s", spFile)
+	spFileContents, err := ioutil.ReadFile(spFile)
+	if err != nil {
+		return nil, err
+	}
+
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.credsSecretName(o),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
```

Comment:
Will check in followup, working on deprovision code, once I feel that's ready to test i'll start spinning up clusters and correct this if necessary.

Reply:
Looking at this more closely, I think it's because for our purposes, we need to mount a single file into the container. If we mounted an entire secret with individual keys, I believe that gets us a directory with multiple files, one per key, which wouldn't work when you try to mount in and run the installer.

Reply:
yeah, i see now that we've been using env vars for AWS.